packages/js/src/generators/typescript-sync/typescript-sync.ts

```diff
@@ -465,13 +465,22 @@ function collectProjectDependencies(
 
   collectedDependencies.set(projectName, []);
 
+  const implicitDependencies = new Set(
+    projectGraph.nodes[projectName].data.implicitDependencies ?? []
+  );
+
   for (const dep of projectGraph.dependencies[projectName]) {
     const targetProjectNode = projectGraph.nodes[dep.target];
     if (!targetProjectNode) {
       // It's an npm dependency
       continue;
     }
 
+    if (implicitDependencies.has(targetProjectNode.name)) {
+      // It's an implicit dependency and therefore not explicitly referenced in code
+      continue;
+    }
```

Comment:
A project could potentially have a static/dynamic dependency on another project and also have it as an implicit dependency. While we could argue that's a bad user configuration, it's currently a possible/valid configuration. In that case, we still want to process the target project if it's also defined as a static/dynamic dependency.
```suggestion
  for (const dep of projectGraph.dependencies[projectName]) {
    const targetProjectNode = projectGraph.nodes[dep.target];
    if (!targetProjectNode || dep.type === 'implicit') {
      // It's an npm or an implicit dependency
      continue;
    }
```

Reply:
@leosvelperez thanks for the review! Great callout - have committed your suggestion :)